TestMauidotnet8/MyPopup.xaml.cs

```diff
@@ -1,13 +1,21 @@
+using CommunityToolkit.Maui.Core;
 using CommunityToolkit.Maui.Views;
 
 namespace TestMauidotnet8;
 
 public partial class MyPopup : Popup
 {
-    public MyPopup(MyViewModel myViewModel)
+    readonly IPopupService popupService;
+    
+    public MyPopup(MyViewModel myViewModel, IPopupService popupService)
     {
         InitializeComponent();
-
+        this.popupService = popupService;
         BindingContext = myViewModel;
     }
+    
+    void HandleButtonClicked(object? sender, EventArgs e)
+    {
+        popupService.ShowPopup((MyViewModel)BindingContext);
```

Comment:
When you _do_ have access to the Popup ViewModel that was instantiated by Dependency Injection, you can reference that instance and pass it into `ShowPopup(TViewModel);`

Reply:
Nobody is going to use it like that.

Reply:
This feedback is not helpful. 

Please help me understand why.

Reply:
I use the onPresenting action to manipulate/populate the viewmodel with its data from service as needed:
`var res = await popupService.ShowPopupAsync<TViewModel>(
    (vm) => { 
        vm.LoadById(id); 
});`